pkg/collector/corechecks/net/network/network.go

```diff
@@ -221,63 +427,275 @@ func submitProtocolMetrics(sender sender.Sender, protocolStats net.ProtoCounters
+			} else {
+				continue
+			}
 
-		counterNames := strings.Split(line[i+2:], " ")
+			counterNames := strings.Split(line[i+2:], " ")
 
-		if !scanner.Scan() {
-			return nil, errors.New("/proc/net/netstat is not fomatted correctly, not data line")
-		}
-		line = scanner.Text()
+			if !scanner.Scan() {
+				return nil, errors.New(procfsPath + "/net/" + subdirectory + " is not fomatted correctly, not data line")
+			}
+			line = scanner.Text()
 
-		counterValues := strings.Split(line[i+2:], " ")
-		if len(counterNames) != len(counterValues) {
-			return nil, errors.New("/proc/net/netstat is not fomatted correctly, expected same number of columns")
+			counterValues := strings.Split(line[i+2:], " ")
+			if len(counterNames) != len(counterValues) {
+				return nil, errors.New(procfsPath + "/net/" + subdirectory + " is not fomatted correctly, expected same number of columns")
+			}
+			for j := range counterNames {
+				value, err := strconv.ParseInt(counterValues[j], 10, 64)
+				if err != nil {
+					return nil, err
+				}
+				counters[protocolName].Stats[counterNames[j]] = value
+			}
 		}
+	}
+	return counters, nil
+}
 
-		for j := range counterNames {
-			value, err := strconv.ParseInt(counterValues[j], 10, 64)
+func readIntFile(filePath string, fs afero.Fs) (int, error) {
+	data, err := afero.ReadFile(fs, filePath)
+	if err != nil {
+		return 0, err
+	}
+	value, err := strconv.Atoi(strings.TrimSpace(string(data)))
+	if err != nil {
+		return 0, err
+	}
+	return value, nil
+}
+
+func addConntrackStatsMetrics(sender sender.Sender, conntrackPath string, useSudoConntrack bool) {
+	cmd := []string{conntrackPath, "-S"}
+	if useSudoConntrack {
+		cmd = append([]string{"sudo"}, cmd...)
+	}
+
+	output, err := runCommandFunction(cmd)
+	if err != nil {
+		log.Debugf("Couldn't use %s to get conntrack stats: %v", conntrackPath, err)
+		return
+	}
+
+	lines := strings.Split(output, "\n")
+	for _, line := range lines {
+		if line == "" {
+			continue
+		}
+		cols := strings.Fields(line)
+		cpuNum := strings.Split(cols[0], "=")[1]
+		cpuTag := []string{"cpu:" + cpuNum}
+		cols = cols[1:]
+
+		for _, cell := range cols {
+			parts := strings.Split(cell, "=")
+			if len(parts) != 2 {
+				continue
+			}
+			metric, valueStr := parts[0], parts[1]
+			valueFloat, err := strconv.ParseFloat(valueStr, 64)
 			if err != nil {
-				return nil, err
+				log.Debugf("Error converting value %s for metric %s: %v", valueStr, metric, err)
+				continue
 			}
-			counters[counterNames[j]] = value
+			sender.MonotonicCount("system.net.conntrack."+metric, valueFloat, "", cpuTag)
 		}
 	}
+}
 
-	return counters, nil
+func runCommand(cmd []string) (string, error) {
+	execCmd := exec.Command(cmd[0], cmd[1:]...)
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	execCmd.Stdout = &out
+	execCmd.Stderr = &stderr
+	err := execCmd.Run()
+	if err != nil {
+		return "", fmt.Errorf("error executing command: %v, stderr: %s", err, stderr.String())
+	}
+	return out.String(), nil
```

Comment:
refactor: it looks like everywhere we call `runCommand(...)` we then split on the lines:

```
	lines := strings.Split(output, "\n")
	for _, line := range lines {	
```
	
I believe the go-idiom for this is to use a [bufio.Scanner](https://pkg.go.dev/bufio#Scanner.Scan) and Scan() per line, so could we leave this returning a io.Reader instead of a string and update the callsites to use a Scanner?

```
    scanner := bufio.NewScanner(output)
    for scanner.Scan() {
        line := scanner.Text()
    }
```